File: julia/pascals-triangle/pascals-triangle.jl
function triangle(n::Int)
  if n < 0
    throw(DomainError())
  end
  if n == 0
    return []
  end
  if n == 1
    return [[1]]
  end
  prev = triangle(n-1)
  lastrow = copy(prev[end])
  push!(lastrow,0)
  unshift!(lastrow,0)
  newrow = []
  for i in 1:(length(lastrow) - 1)
    push!(newrow, lastrow[i] + lastrow[i+1])
  end
  push!(prev, newrow)
  return prev
end

File: go/pascals-triangle/pascals_triangle.go
package pascal

const testVersion = 1

func Triangle(n int) [][]int {
	if n == 0 {
		return [][]int{{}}
	}
	if n == 1 {
		return [][]int{{1}}
	}
	prev := Triangle(n - 1)
	last_row := prev[len(prev)-1]
	new_row := []int{1}
	for i := 0; i < len(last_row)-1; i++ {
		new_row = append(new_row, last_row[i]+last_row[i+1])
	}
	new_row = append(new_row, 1)
	triangle := append(prev, new_row)
	return triangle
}
